datafusion/src/execution/options.rs

```diff
@@ -0,0 +1,173 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//! User facing options for the file formats readers
+
+use std::sync::Arc;
+
+use arrow::datatypes::{Schema, SchemaRef};
+
+use crate::datasource::{
+    file_format::{avro::AvroFormat, csv::CsvFormat},
+    listing::ListingOptions,
+};
+
+/// CSV file read option
+#[derive(Copy, Clone)]
+pub struct CsvReadOptions<'a> {
```

Comment:
I wonder if it would be better to put these files into the same modules as the file format that uses them: `file_format::avro::Avro` etc

Reply:
Yes I was kind of hesitant about this! These option struct are really meant to be specific to `context.read_xxx` and `context.register_xxx` APIs, to serve as documentation and default holders. I will probably do a separate PR to propose something a bit more coherent.